pypeit/pypeit.py

```diff
@@ -514,12 +517,11 @@ def reduce_exposure(self, frames, bg_frames=None, std_outfile=None):
             msgs.info(bg_msgs_string)
 
         # Find the detectors to reduce
-#        detectors = PypeIt.select_detectors(detnum=self.par['rdx']['detnum'],
-#                                            slitspatnum=self.par['rdx']['slitspatnum'],
-#                                            ndet=self.spectrograph.ndet)
         subset = self.par['rdx']['slitspatnum'] if self.par['rdx']['slitspatnum'] is not None \
                     else self.par['rdx']['detnum']
         detectors = self.spectrograph.select_detectors(subset=subset)
+        # Allow for multiple slitspatnum entries
+        detectors = np.unique(detectors).tolist()
```

Comment:
This is where things go wrong for the Gemini GMOS mosaic.  I.e.:

```
In [1]: subset
Out[1]: [(1, 2, 3)]

In [2]: self.spectrograph.select_detectors(subset=subset)
Out[2]: [(1, 2, 3)]

In [3]: np.unique(self.spectrograph.select_detectors(subset=subset)).tolist()
Out[3]: [1, 2, 3]
```

I haven't dug in enough to understand the need for this line, but this won't work when `detectors` is a tuple listing the detectors in a mosaic.  I'm happy to chat about how to handle this, but I would start by moving what you're trying to do inside the `select_detectors` function.

Reply:
This does it:  `return list(set(_subset))`
 (moved to `select_detectors`)